Sources/Firebase/Models/Requests/DynamicLinkRequest.swift

```diff
@@ -0,0 +1,176 @@
+        public let androidMinPackageVersionCode: String?
+
+        public init(packageName: String,
+                    fallbackLink: String? = nil,
+                    minPackageVersionCode: String? = nil) {
+
+            self.androidPackageName = packageName
+            self.androidFallbackLink = fallbackLink
+            self.androidMinPackageVersionCode = minPackageVersionCode
+        }
+    }
+
+    public struct Ios: FirebaseRequestModel {
+
+        public let iosBundleId: String
+        public let iosAppStoreId: String
+        public let iosFallbackLink: String?
+        public let iosCustomScheme: String?
+        public let iosIpadFallbackLink: String?
+        public let iosIpadBundleId: String?
+
+        public init(bundleId: String,
+                    appStoreId: String,
+                    fallbackLink: String? = nil,
+                    customScheme: String? = nil,
+                    ipadFallbackLink: String? = nil,
+                    ipadBundleId: String? = nil) {
+
+            self.iosBundleId = bundleId
+            self.iosAppStoreId = appStoreId
+            self.iosFallbackLink = fallbackLink
+            self.iosCustomScheme = customScheme
+            self.iosIpadFallbackLink = ipadFallbackLink
+            self.iosIpadBundleId = ipadBundleId
+        }
+    }
+
+    public struct Navigation: FirebaseRequestModel {
+
+        public let enableForcedRedirect: Bool
+
+        public init(enableForcedRedirect: Bool) {
+            self.enableForcedRedirect = enableForcedRedirect
+        }
+    }
+
+    public struct Analytics: FirebaseRequestModel {
+
+        public let googlePlayAnalytics: GooglePlay?
+        public let itunesConnectAnalytics: ItunesConnect?
+
+        public init(googlePlay: GooglePlay? = nil,
+                    itunesConnect: ItunesConnect? = nil) {
+            self.googlePlayAnalytics = googlePlay
+            self.itunesConnectAnalytics = itunesConnect
+        }
+
+        public struct GooglePlay: FirebaseRequestModel {
+
+            public let utmSource: String
+            public let utmMedium: String
+            public let utmCampaign: String
+            public let utmTerm: String
+            public let utmContent: String
+            public let gclid: String
+
+            public init(utmSource: String,
+                        utmMedium: String,
+                        utmCampaign: String,
+                        utmTerm: String,
+                        utmContent: String,
+                        gclid: String) {
+
+                self.utmSource = utmSource
+                self.utmMedium = utmMedium
+                self.utmCampaign = utmCampaign
+                self.utmTerm = utmTerm
+                self.utmContent = utmContent
+                self.gclid = gclid
+            }
+        }
+
+        public struct ItunesConnect: FirebaseRequestModel {
+
+            public let at: String
+            public let ct: String
+            public let mt: String
```

Comment:
you should rename this `mediaType` and you can also convert it into an enum like this:

```swift
enum ITCMediaType: Int {
    case music = 1
    case podcasts = 2
    case audiobooks = 3
    case tvShows = 4
    case musicVideos = 5
    case movies = 6
    case iPodGames = 7
    case mobileSoftwareApplications = 8
    case ringtones = 9
    case itunesU = 10
    case eBooks = 11
    case desktopApps = 12
}
```

Reply:
https://github.com/h2glab/firebase-provider/pull/3/commits/d6d756dc72192a0d4d003929589046a443df4802